gufe/tests/test_protocol.py

```diff
@@ -226,7 +226,7 @@ def test_dag_execute(self, protocol_dag):
         assert finishresult.name == "the end"
 
         # gather SimulationUnits
-        simulationresults = [dagresult.unit_to_result(pu)
+        simulationresults = [dagresult.unit_to_results(pu)[0]
```

Comment:
`unit.to_result` is renamed `unit_to_results` as there are now many results per Unit.  This is slightly ugly, as more often than not, people will be interested in the success Result unless they explicitly ask for *all* results.

Perhaps I could keep `unit_to_result` (singular) and make this always return the corresponding `.ok()` Result, then `unit_to_all_results` would give the list of multiple possible Results (some fails possible)

Reply:
> Perhaps I could keep `unit_to_result` (singular) and make this always return the corresponding `.ok()` Result, then `unit_to_all_results` would give the list of multiple possible Results (some fails possible)

I like this proposal. I think it has the characteristics we want here.